File: activity_plugin_platform_interface/lib/src/model/data_type.dart
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

import 'package:flutter/foundation.dart';

enum DataType {
  steps,
}

extension DataTypeExtension on DataType {
  String toJson() {
    return describeEnum(this);
  }
}

File: activity_plugin/android/src/main/kotlin/uk/co/objectivity/activity_plugin/model/DataType.kt
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

package uk.co.objectivity.activity_plugin.model

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

enum class DataType(val rawName: String) {
    STEPS("steps");

    class Adapter {
        @FromJson
        fun fromJson(json: String): DataType = values().find { it.rawName == json }!!

        @ToJson
        fun toJson(value: DataType): String = value.rawName
    }
}
